games/rpg/src/lib.rs

```diff
@@ -0,0 +1,1483 @@
+                new_accessory
+            }
+            else if rng == 2 {
+                let accessory_data = Accessory {  
+                    id: dec!(1),
+                    part: part,
+                    level: dec!(1), 
+                    attack: atk,
+                    magic: mag_bonus,
+                    speed: spd,
+                    weight: load,
+                    ability: dec!(0),
+                    ability_scaling: ability_scale,
+                    version: dec!(1),
+                };
+                let new_accessory = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.item_nft)
+                        .mint_non_fungible(&NonFungibleId::random(), accessory_data));
+                
+                self.system_vault.authorize(|| 
+                    gold.burn());
+                self.system_vault.authorize(||
+                    resource1.burn());
+                self.system_vault.authorize(||
+                    resource2.burn());
+                self.system_vault.put(key_bucket);
+                new_accessory
+            }
+            else {
+                let accessory_data = Accessory {  
+                    id: dec!(1),
+                    part: part,
+                    level: dec!(1), 
+                    attack: atk,
+                    magic: mag,
+                    speed: spd_bonus,
+                    weight: load,
+                    ability: dec!(0),
+                    ability_scaling: ability_scale,
+                    version: dec!(1),
+                };
+                let new_accessory = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.item_nft)
+                        .mint_non_fungible(&NonFungibleId::random(), accessory_data));
+                
+                self.system_vault.authorize(|| 
+                    gold.burn());
+                self.system_vault.authorize(||
+                    resource1.burn());
+                self.system_vault.authorize(||
+                    resource2.burn());
+                self.system_vault.put(key_bucket);
+                new_accessory
+            }
+        }
+        // Place character,weapon,armor, and accessory data + stage # to fight. 
+        // Method calculates whether you win and grants rewards based on win or loss
+        // example command: resim call-method $c stage 1,$char "#$weapon,#$helmet,#$chest,#$pants,#$gloves,#$belt,#$shoes,$item" 1
+        pub fn stage(&mut self, 
+            nft_proof: Proof, 
+            gear: Proof, 
+            stage: u64,
+            ) -> (Bucket, Bucket, Bucket) {
+            // Takes system badge for authorization
+            let key_bucket: Bucket = self.system_vault.take(1);
+            // Data from Proofs
+            let mut nft_data: Account = nft_proof.non_fungible().data();
+            let weapon_data: Weapon = gear.non_fungibles()[0].data();
```

Comment:
It feels to me like this method can fail quite often since the order of the items in the `Proof` is hard-coded into the code. This is not how it should be done in Scrypto. 

In my previous comments, I recommended that you split up the three different types of NFTs you have into multiple resources. If you do so, you will no longer need to assume the position of each one of the items in the proof. You would simply have this method take three `Proof`s in a tuple, vector, or as three separate arguments. The three `Proof`s would belong to the three types of items you have.